craft_application/application.py

```diff
@@ -366,6 +369,29 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
 
         return dispatcher
 
+    def _get_app_plugins(self) -> dict[str, PluginType]:
+        """Get the plugins for this application.
+
+        Should be overridden by applications that need to register plugins at startup.
+        """
+        return {}
+
+    def register_plugins(self, plugins: dict[str, PluginType]) -> None:
+        """Register plugins for this application."""
+        if not plugins:
+            return
+
+        craft_cli.emit.trace("Registering plugins...")
+        craft_cli.emit.trace(f"Plugins: {', '.join(plugins.keys())}")
+        craft_parts.plugins.register(plugins)
+
+    def _register_default_plugins(self) -> None:
+        """Register default + per application plugins when initializing."""
+        default_plugins: dict[str, PluginType] = {}
```

Comment:
what's the purpose of this empty `default_plugins` dict?

Reply:
Just in case if craft-application want to include some plugins. If that is very unlikely, I can remove that.

Reply:
Let's remove it for now and re-add it later if we do create any plugins in craft-application. Right now I can't see a specific use case